feature/experimental/bgp/ate_tests/bgp_2byte_4byte_asn/bgp_2byte_4byte_asn_test.go

```diff
@@ -0,0 +1,248 @@
+			bgpClearConfig(t, dut)
+
+			t.Log("Configure BGP on DUT")
+			gnmi.Replace(t, dut, dutConfPath.Config(), tc.dutConf)
+
+			fptest.LogQuery(t, "DUT BGP Config ", dutConfPath.Config(), gnmi.GetConfig(t, dut, dutConfPath.Config()))
+			t.Log("Configure BGP on ATE")
+			tc.ateConf.Push(t)
+			tc.ateConf.StartProtocols(t)
+
+			t.Log("Verify BGP session state : ESTABLISHED")
+			nbrPath := statePath.Neighbor(tc.nbr.peerIP)
+			gnmi.Await(t, dut, nbrPath.SessionState().State(), time.Second*60, oc.Bgp_Neighbor_SessionState_ESTABLISHED)
+
+			t.Log("Verify BGP AS numbers")
+			verifyPeer(t, &tc.nbr, dut)
+
+			t.Log("Clear BGP Configs on ATE")
+			tc.ateConf.StopProtocols(t)
+		})
+	}
+}
+
+// bgpClearConfig removes all BGP configuration from the DUT.
+func bgpClearConfig(t *testing.T, dut *ondatra.DUTDevice) {
+
+	resetBatch := &gnmi.SetBatch{}
+	gnmi.BatchDelete(resetBatch, gnmi.OC().NetworkInstance(deviations.DefaultNetworkInstance(dut)).Protocol(oc.PolicyTypes_INSTALL_PROTOCOL_TYPE_BGP, "BGP").Config())
+
+	if deviations.NetworkInstanceTableDeletionRequired(dut) {
+		tablePath := gnmi.OC().NetworkInstance(deviations.DefaultNetworkInstance(dut)).TableAny()
+		for _, table := range gnmi.LookupAll(t, dut, tablePath.Config()) {
+			if val, ok := table.Val(); ok {
+				if val.GetProtocol() == oc.PolicyTypes_INSTALL_PROTOCOL_TYPE_BGP {
+					gnmi.BatchDelete(resetBatch, gnmi.OC().NetworkInstance(deviations.DefaultNetworkInstance(dut)).Table(oc.PolicyTypes_INSTALL_PROTOCOL_TYPE_BGP, val.GetAddressFamily()).Config())
+				}
+			}
+		}
+	}
+	resetBatch.Set(t, dut)
+}
+
+func verifyPeer(t *testing.T, nbr *bgpNbr, dut *ondatra.DUTDevice) {
+
+	statePath := gnmi.OC().NetworkInstance(deviations.DefaultNetworkInstance(dut)).Protocol(oc.PolicyTypes_INSTALL_PROTOCOL_TYPE_BGP, "BGP").Bgp()
+	nbrPath := statePath.Neighbor(nbr.peerIP)
+	glblPath := statePath.Global()
+
+	// Check BGP peerAS from telemetry.
+	peerAS := gnmi.Get(t, dut, nbrPath.State()).GetPeerAs()
+	if peerAS != nbr.peerAS {
+		t.Errorf("BGP peerAs: got %v, want %v", peerAS, nbr.peerAS)
+	}
+
+	// Check BGP localAS from telemetry.
+	localAS := gnmi.Get(t, dut, nbrPath.State()).GetLocalAs()
+	if localAS != nbr.localAS {
+		t.Errorf("BGP localAS: got %v, want %v", localAS, nbr.localAS)
+	}
+
+	// Check BGP globalAS from telemetry.
+	globalAS := gnmi.Get(t, dut, glblPath.State()).GetAs()
+	if globalAS != nbr.localAS {
+		t.Errorf("BGP globalAS: got %v, want %v", globalAS, nbr.localAS)
+	}
+}
+
+func configureATE(t *testing.T, ateParams *bgpNbr, connectionType string) *ondatra.ATETopology {
+
+	t.Log("Configure ATE interface")
+	ate := ondatra.ATE(t, "ate")
+	port1 := ate.Port(t, "port1")
+	topo := ate.Topology().New()
+
+	iDut1 := topo.AddInterface(ateSrc.Name).WithPort(port1)
+	iDut1.IPv4().WithAddress(ateSrc.IPv4CIDR()).WithDefaultGateway(dutSrc.IPv4)
+	iDut1.IPv6().WithAddress(ateSrc.IPv6CIDR()).WithDefaultGateway(dutSrc.IPv6)
+
+	bgpDut1 := iDut1.BGP()
+
+	if connectionType == connInternal {
```

Comment:
create peer outside the if-statement and only set internal/external inside the if/else

```
peer := iDut1.BGP().AddPeer().WithPeerAddress(ateParams.peerIP).WithLocalASN(ateParams.localAS)
if connectionType == connInternal {
  peer.WithTypeInternal()
} else {
  peer.WithTypeExternal()
}
```